urllib3/util/selectors.py

```diff
@@ -535,19 +535,44 @@ def close(self):
             super(KqueueSelector, self).close()
 
 
+if not hasattr(select, 'select'):  # Platform-specific: AppEngine
+    HAS_SELECT = False
+
+
+def _can_allocate(struct):
+    """ Checks that select structs can be allocated
+    by the underlying operating system, not just
+    advertised by the select module. We don't check
+    select() because we'll be hopeful that most 
+    platforms that don't have it available will not
+    advertise it. (ie: GAE) """
+    try:
+        getattr(select, struct)().close()
```

Comment:
So this definitely won't work, will have to detect if the selector has a `close()` function and use it then. (`select.poll` has no `close()`).
Also might have to add a scenario for `select.poll.poll()` as it won't throw the error until it's actually used.